### PR TITLE
Handle unpredictable GitHub timeouts

### LINKED ISSUE #28: Peloton is timing out
Have done some rudimentary investigation with the help of @stephenworsley and @ESadek-MO. By experimenting with `PaginatedMutation._PAGINATION`:

https://github.com/SciTools/.github/blob/6cd87da89ae606ccb4ce0a4c95421c012cb29108/peloton/update_project.py#L846-L854

... we have so far confirmed:

- This is a timeout problem - failures consistently occur after 10sec, which we believe is GitHub's chosen timeout[^1]; dropping `_PAGINATION` to `1` gets each query under 10secs and then consistently succeeds.
- It is specific to the [`updateProjectV2ItemFieldValue`](https://docs.github.com/en/graphql/reference/mutations#updateprojectv2itemfieldvalue) mutation - this is the last stage of the script; all previous mutation steps (e.g. adding, deleting, field-clearing) consistently succeed with the mutation pagination still set to `20`.
- It is probably an issue with overhead - dropping `_PAGINATION` to `10` produced queries running in around 10secs, dropping to `1` only got down to 6secs. Given GitHub's the apparent 10sec timeout[^1], a 5-6sec overhead doesn't leave much wiggle room.
- There are no recent reports of problems like this.

Using a `_PAGINATION` of `1` is not a good solution - updates will take prohibitively long, especially when recovering from an outage like this (more updates to do). A value of `10`, coupled with some sort of repeat pattern using `try`-`except` might be the way to go. We agreed it would be worth reporting a MWE - including timings - to GitHub.

For now, the GitHub Action remains disabled. I have managed to update the Peloton by running locally with a `_PAGINATION` of `1`. Anyone can run locally using an appropriate personal access token:

https://github.com/SciTools/.github/blob/6cd87da89ae606ccb4ce0a4c95421c012cb29108/peloton/update_project.py#L1202-L1209

[^1]: The documentation makes reference to this, although is a little unclear, and it is certainly consistent with what we've seen.

### PR DESCRIPTION
**Pull request by trexfeathers**
Closes #28

It would appear that the situation described in #28 has been resolved, but I have added this code to help cope with future problems.

### CODE REVIEW

**Comment by trexfeathers** [comment]
Going to start as a draft while I test that this works (I needed to raise the PR so that there were changes to update on the Peloton!).

File: peloton/update_project.py [hunk 1]
```diff
@@ -928,10 +928,31 @@ def _run_sub_mutation(self, page_slice: slice) -> list:
...
-        result = run_operation(mutation)
-        # Fields are accessed from mutation results via their keys.
-        payloads = [result[a] for a in self._aliases[page_slice]]
-        return payloads
+
+        # Handle unpredictable GitHub timeouts.
+        attempts = 0
+        result = None
+        exception = None
+        while attempts < 5:
```
Comment (stephenworsley) [inline peloton/update_project.py:936]: Couldn't this be a for loop? Is there a reason to keep hold of the number of attempts (debugging maybe)? If so, I feel like you'd get a more accurate number by incrementing `attempts` before the try except block.

**Review by stephenworsley — CHANGES_REQUESTED** [review]
Looking good, just a short question on the `while` loop.

**Review by stephenworsley — APPROVED** [review]
Looks good.

**Pull request merged**